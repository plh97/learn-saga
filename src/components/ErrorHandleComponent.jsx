import React from 'react'

const initState = {
  message: '',
  stack: '',
}

export default function({ data = initState }) {
  return (
    <div className="error">
      <div className="message">{data.message}</div>
      <div className="stack">{data.stack}</div>
    </div>
  )
}