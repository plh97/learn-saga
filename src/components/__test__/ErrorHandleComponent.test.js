import React from 'react'
import ErrorHandleComponent from '../ErrorHandleComponent'
import renderer from 'react-test-renderer'


// LabelItem 组件 未选中 状态
describe('Loading snapshot rending test', () => {
  it('renders correctly with default props', () => {
    const component = renderer
      .create(<ErrorHandleComponent />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  it('renders correctly with my setting props', () => {
    const data = {
      message: '加载中',
      stack: 'sdfsad',
    }
    const component = renderer
      .create(<ErrorHandleComponent data={data} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
