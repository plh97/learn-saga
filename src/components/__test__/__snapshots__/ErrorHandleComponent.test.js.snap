// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshot rending test renders correctly with default props 1`] = `
<div
  className="error"
>
  <div
    className="message"
  >
    
  </div>
  <div
    className="stack"
  >
    
  </div>
</div>
`;

exports[`Loading snapshot rending test renders correctly with my setting props 1`] = `
<div
  className="error"
>
  <div
    className="message"
  >
    加载中
  </div>
  <div
    className="stack"
  >
    sdfsad
  </div>
</div>
`;
